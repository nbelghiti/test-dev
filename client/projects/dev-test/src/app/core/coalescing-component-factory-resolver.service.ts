/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
