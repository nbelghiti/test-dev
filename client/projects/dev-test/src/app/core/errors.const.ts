export const ToastrConfig = {
	maxOpened: 2,
	timeOut: 8000
};
