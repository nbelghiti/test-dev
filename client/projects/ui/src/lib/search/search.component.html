<div class="c-search l-container">
	<ng-content select="ui-search-field"></ng-content>
	<ng-content select="ui-search-content"></ng-content>

	<div class="c-search__overlay" (click)="clicked.emit(false)"></div>
</div>
