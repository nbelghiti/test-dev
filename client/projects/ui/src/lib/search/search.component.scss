// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Search
// =============================================================================
.c-search {
	position: absolute;
	top: 2rem;
	left: 0;
	right: 0;
	z-index: $z-index-atmosphere;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-search__overlay {
	position: fixed;
	top: 0;
	left: 0;
	bottom: 0;
	right: 0;
	z-index: $z-index-pit;

	background-color: $black-50;
}
