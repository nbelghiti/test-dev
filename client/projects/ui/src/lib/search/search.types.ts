export interface ISearchQuery {
	query?: string;
	in?: string;
	project?: string;
	package?: string;
	group?: string;
	component?: string;
}

export enum SearchResultType {
	project = 'project',
	package = 'package',
	group = 'group',
	component = 'component',
	field = 'field',
}

export interface ISearchResultItem {
	id: string;
	type: SearchResultType;
	description: string;
	project?: {
		id: string;
		name: string;
	};
	package?: {
		id: string;
		name: string;
	};
	componentGroup?: {
		id: string;
		name: string;
	};
	component?: {
		id: string;
		name: string;
	};
	field?: {
		id: string;
		name: string;
	};
	path: string;
	link: string[];
}
