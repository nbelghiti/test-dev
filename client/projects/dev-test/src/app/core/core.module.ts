import { ErrorHandler, NgModule } from '@angular/core';

import { CoalescingComponentFactoryResolver } from './coalescing-component-factory-resolver.service';
import { ContextModule } from './context/context.module';
import { GlobalErrorHandler } from './global-error-handler.service';
import { RouterModule } from './router/router.module';
import { UserModule } from './user/user.module';

@NgModule({
	imports: [
		ContextModule,
		RouterModule,
		UserModule,
	],
	providers: [
		CoalescingComponentFactoryResolver,
		{
			provide: ErrorHandler,
			useClass: GlobalErrorHandler,
		},
	],
})
export class CoreModule {
	constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
		coalescingResolver.init();
	}
}
