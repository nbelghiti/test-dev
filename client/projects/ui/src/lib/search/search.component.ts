import { Component, EventEmitter, Output } from '@angular/core';

@Component({
	selector: 'ui-search',
	templateUrl: './search.component.html',
	styleUrls: ['./search.component.scss'],
})

export class SearchComponent {
	@Output() public clicked: EventEmitter<boolean> = new EventEmitter<boolean>();
}
