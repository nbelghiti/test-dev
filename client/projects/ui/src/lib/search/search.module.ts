import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';

import { EmptyStateModule } from '../empty-state/empty-state.module';
import { IconModule } from '../icon/icon.module';

import { SearchContentComponent } from './content/search-content.component';
import { SearchFieldComponent } from './field/search-field.component';
import { SearchGroupComponent } from './group/search-group.component';
import { SearchIndicatorComponent } from './indicator/search-indicator.component';
import { SearchQueryComponent } from './query/search-query.component';
import { SearchResultComponent } from './result/search-result.component';
import { SearchResultsComponent } from './results/search-results.component';
import { SearchComponent } from './search.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
		FormsModule,
		ReactiveFormsModule,
		EmptyStateModule,
	],
	declarations: [
		SearchComponent,
		SearchContentComponent,
		SearchFieldComponent,
		SearchGroupComponent,
		SearchIndicatorComponent,
		SearchQueryComponent,
		SearchResultComponent,
		SearchResultsComponent,
	],
	exports: [
		SearchComponent,
		SearchContentComponent,
		SearchFieldComponent,
		SearchGroupComponent,
		SearchIndicatorComponent,
		SearchQueryComponent,
		SearchResultComponent,
		SearchResultsComponent,
	],
})
export class SearchModule {}
