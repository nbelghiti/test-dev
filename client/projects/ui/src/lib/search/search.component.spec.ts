import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

import { SearchComponent } from './search.component';

describe('SearchComponent', () => {
	let component: SearchComponent;
	let fixture: ComponentFixture<SearchComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [ SearchComponent ],
			imports: [
				FormsModule,
				ReactiveFormsModule,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(SearchComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
