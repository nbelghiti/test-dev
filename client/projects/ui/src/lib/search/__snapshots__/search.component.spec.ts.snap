// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent Should render correctly 1`] = `
<ui-search
  clicked={[Function EventEmitter]}
>
  <div
    class="c-search l-container"
  >
    <div
      class="c-search__overlay"
    />
  </div>
</ui-search>
`;
